packages/toast/toast-container.js

```diff
@@ -0,0 +1,138 @@
+            }, 500);
+        }
+    }
+    
+    disconnectedCallback() {
+        super.disconnectedCallback();
+        if (this._interval) clearTimeout(this._interval);
+    }
+
+    static init() {
+        let el = document.querySelector('f-toast-container');
+        if (!el) {
+            el = document.createElement('f-toast-container');
+            document.body.appendChild(el);
+        }
+        return el;
+    }
+
+    get _toastsArray() {
+        return Array.from(this._toasts).map(([,toast]) => toast);
+    }
+
+    /**
+     * 
+     * @param {String|Number} id 
+     * @returns {ToastOptions}
+     */
+    get(id) {
+        if (!id) throw new Error('undefined "id" given when attempting to retrieve toast');
+        if (typeof id !== 'string' && !Number.isInteger(id)) throw new Error('"id" must be number or string when attempting to retrieve toast');
+        return this._toasts.get(id);
+    }
+
+    /**
+     * 
+     * @param {Object} toast 
+     * @returns {FabricToastContainer}
+     */
+    set(toast) {
+        if (!toast.id) throw new Error('invalid or undefined "id" on toast object');
+        const result = this._toasts.set(toast.id, {
+            ...toast,
+            duration: Date.now() + toast.duration || Number.POSITIVE_INFINITY,
+        });
+        this._toasts = new Map(Array.from(this._toasts));
+        return result;
+    }
+
+    /**
+     * 
+     * @param {String|Number} id
+     * @returns {ToastOptions}
+     */
+    async del(id) {
+        if (!id) throw new Error('undefined "id" given when attempting to retrieve toast');
+        if (typeof id !== 'string' && !Number.isInteger(id)) throw new Error('"id" must be number or string when attempting to retrieve toast');
+        const el = this.renderRoot.querySelector(`#${id}`);
+        await el.collapse();
+        const result = this._toasts.delete(id);
+        this._toasts = new Map(Array.from(this._toasts));
+        return result;
+    }
+
+    render() {
+        return html`
+            <link
+                rel="stylesheet"
+                type="text/css"
+                href="https://assets.finn.no/pkg/@fabric-ds/css/v1/fabric.min.css"
+            />
+            <aside class="${c.toasterContainer}">
+                <div class="${c.toaster}" id="f-toast-container-list"> 
+                    ${repeat(this._toastsArray, (toast) => toast.id, (toast) => html`
+                    <f-toast
+                        class="w-full"
+                        id="${toast.id}"
+                        type="${toast.type}"
+                        text="${toast.text}"
+                        ?canclose=${toast.canclose === false ? false : true }
+                        @close=${() => this.del(toast.id)}>
+                    </f-toast>`)}
+                </div>
+            </aside>
+        `;
+    }
+}
+
+if (!customElements.get('f-toast-container')) {
+    customElements.define('f-toast-container', FabricToastContainer);
+}
```

Comment:
The FabricToastContainer essentially does 3 things:

1. Holds internal map of toast data
2. Provides public API methods for managing the set of toasts (get,set and del)
3. Renders out a list of toasts whenever the internal state changes